services/venice-router/src/main/java/com/linkedin/venice/router/throttle/NoopRouterThrottler.java

```diff
@@ -5,25 +5,43 @@
 import com.linkedin.venice.meta.RoutersClusterManager;
 import com.linkedin.venice.meta.Store;
 import com.linkedin.venice.meta.StoreDataChangedListener;
+import com.linkedin.venice.router.VeniceRouterConfig;
 import com.linkedin.venice.router.stats.AggRouterHttpRequestStats;
 import java.util.List;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 
 public class NoopRouterThrottler
```

Comment:
This class seems not designed properly..
If we want to keep the same store-level quota as the regular `ReadRequestThrottler`, why not just introduces a feature flag in `ReadRequestThrottler` to indicate whether the throttler is a pass-through or not, and we don't need this `NoOpRouterThrottler` at all.
By eliminating this class, we can continue to construct a `ReadRequestThrottler` in constructor function and change the feature flag at runtime and this will help address the previous comment about potential leaking of ZK watches as well.

What do you think?

Reply:
yeah I mentioned in approach in the slack. For noop throttler, reusing the read throttler code might be bit heavy when it really not need anything but emitting the metrics.

Reply:
That doesn't change the fact that  there could be leaking no matter the total number of times.

I think this throttler is mainly used by Dyno testing.

If it can simplify the structure significantly, I don't see much of overhead since `mayThrottleRead` is the only hot path.

Reply:
yeah, code could be simpler. I will remove the noop class to consolidate into `ReadRequestThrottler`